test/Datadog.Trace.DuckTyping.Tests/ReverseProxyTests.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        public enum LogEventLevel
+        {
+            Verbose,
+            Debug,
+            Information,
+            Warning,
+            Error,
+            Fatal
+        }
+
+        public interface IMessageTemplate
+        {
+            public string Text { get; }
+        }
+
+        public interface ILogEventPropertyFactory
+        {
+            ILogEventProperty CreateProperty(string name, object value, bool destructureObjects = false);
+        }
+
+        public interface ILogEventProperty
+        {
+            public string Name { get; }
+
+            public object Value { get; }
+        }
+
+        // ************************************************************************************
+
+        [Fact]
+        public void AbstractClassReverseProxyTest()
+        {
+            var resetEvent = new ManualResetEventSlim();
+
+            var eventInstance = new LogEventPropertyValueImpl(resetEvent);
+
+            var type = typeof(Datadog.Trace.Vendors.Serilog.Events.LogEventPropertyValue);
+#if NET452
+            Assert.Throws<DuckTypeTypeIsNotPublicException>(() =>
+            {
+                eventInstance.DuckCast(type);
+            });
+#else
+            var proxy2 = eventInstance.DuckCast(type);
+            eventInstance.SetBaseInstance(proxy2);
+
+            ((Datadog.Trace.Vendors.Serilog.Events.LogEventPropertyValue)proxy2).ToString("Hello world", null);
+
+            Assert.True(resetEvent.Wait(5_000));
+#endif
+        }
+
+        // ************************************************************************************
+        // Types for AbstractClassReverseProxyTest
+        // ***
+
+        public class LogEventPropertyValueImpl
+        {
+            private IBaseClass _base;
+            private ManualResetEventSlim _manualResetEventSlim;
+
+            public LogEventPropertyValueImpl(ManualResetEventSlim manualResetEventSlim)
+            {
+                _manualResetEventSlim = manualResetEventSlim;
+            }
+
+            public void SetBaseInstance(object baseObject)
+            {
+                _base = baseObject.DuckCast<IBaseClass>();
+            }
+
+            [DuckReverseMethod]
+            public void Render(TextWriter output, string format = null, IFormatProvider formatProvider = null)
+            {
+                output.WriteLine(format);
+
+                Assert.NotNull(output);
+                Assert.Equal("Hello world", format);
+
+                _manualResetEventSlim.Set();
+            }
+
+            public interface IBaseClass
+            {
+                public string ToString();
+
+                public string ToString(string format, IFormatProvider formatProvider);
+            }
+        }
+
+        // ************************************************************************************
+
+        [Fact]
+        public void PublicInterfaceReverProxyTest()
+        {
+            var instance = new PublicBaseInterface();
```

Comment:
Could you do the same thing as the private interface test, but with the "real" serilog instead of the vendored version? This way we'll be completely sure that it works in the intended scenario

Reply:
done in 2e0237f